include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -152,145 +172,183 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
-									const auto outDataAdressOffsetScratch = scratchInPlaneOffset * currentChannelCount;
-
-									auto* entryScratchAdress = scratchMemory + outDataAdressOffsetScratch + singleScratchLayerOffset * w;
-									auto* entryTotalImageCacheValues = reinterpret_cast<decodeType*>(totalImageCacheScratch + (localCoord.w * state->extent.depth + localCoord.z) * state->mainRowScratchCacheByteSize);
-
-									for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
-										entryScratchAdress[channel] /= entryTotalImageCacheValues[channel];
-								}
-				};
 
-				if (state->normalizeImageByTotalSATValues)
-					if (!asset::isNormalizedFormat(inFormat))
-						normalizeScratch();
+					/*
 
-				auto perOutputRegionEncode = [&](const CommonExecuteData& commonExecuteData, CBasicImageFilterCommon::clip_region_functor_t& clip) -> bool
-				{
-					auto encode = [&](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos) -> void
+					auto normalizeScratch = [&]()
 					{
-						const auto scratchInPlaneOffset = ((readBlockPos.z * state->extent.height + readBlockPos.y) * state->extent.width + readBlockPos.x);
+						core::vector3du32_SIMD localCoord;
+							for (auto& z = localCoord[2] = 0u; z < extent.z; ++z)
+								for (auto& y = localCoord[1] = 0u; y < extent.y; ++y)
+									for (auto& x = localCoord[0] = 0u; x < extent.x; ++x)
+									{
+										const auto scratchInPlaneOffset = ((localCoord.z * state->extent.height + localCoord.y) * state->extent.width + localCoord.x);
+										const auto outDataAdressOffsetScratch = scratchInPlaneOffset * currentChannelCount;
+
+										auto* entryScratchAdress = scratchMemory + outDataAdressOffsetScratch;
+										auto* entryTotalImageCacheValues = reinterpret_cast<decodeType*>(totalImageCacheScratch + (localCoord.w * state->extent.depth + localCoord.z) * state->mainRowScratchCacheByteSize);
+
+										for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+											entryScratchAdress[channel] /= entryTotalImageCacheValues[channel];
+									}
+					};
+					*/
 
-						auto* outDataAdress = commonExecuteData.outData + readBlockArrayOffset;
-						const auto outDataAdressOffsetScratch = scratchInPlaneOffset * currentChannelCount;
+					/*
 
-						auto* entryScratchAdress = scratchMemory + outDataAdressOffsetScratch + singleScratchLayerOffset * readBlockPos.w;
-						asset::encodePixelsRuntime(outFormat, outDataAdress, entryScratchAdress); // overrrides texels, so region-overlapping case is fine
+					it wont work anymore since now
 
-						if (readBlockPos.x == trueOutExtent.width - 1 && readBlockPos.y == 0) 
-							memcpy(totalImageCacheOutput + (readBlockPos.w * state->extent.depth + readBlockPos.z) * outTexelByteSize, outDataAdress, outTexelByteSize);
-					};
+					if (state->normalizeImageByTotalSATValues)
+						if (!asset::isNormalizedFormat(inFormat))
```

Comment:
STILL

Reply:
I know I know, I was waiting for review and https://github.com/buildaworldnet/IrrlichtBAW/pull/497#discussion_r463973877